### PR TITLE
Updated WindowsResourceComple to use provider api

### PR DESCRIPTION
**Pull request by fletcher-sumglobal**
Issue [#173](https://github.com/gradle/gradle-native/issues/173) Windows Resource Support
Updated this class to use ListProperty<String> instead of List<String> for the compilerArgs, consistent with other tasks within 'cpp-application'

Signed-off-by: Kent Fletcher <kfletcher@sumglobal.com>

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/language-native/src/main/java/org/gradle/language/rc/tasks/WindowsResourceCompile.java [hunk 1]
```diff
@@ -201,15 +202,17 @@ public void setMacros(Map<String, String> macros) {
...
+     *
+     * @since 5.1
      */
     @Input
-    public List<String> getCompilerArgs() {
+    public ListProperty<String> getCompilerArgs() {
         return compilerArgs;
     }
 
     public void setCompilerArgs(List<String> compilerArgs) {
```
Comment (big-guy) [inline subprojects/language-native/src/main/java/org/gradle/language/rc/tasks/WindowsResourceCompile.java]: It's OK to remove this method.  We just need to change `WindowsResourcesCompileTaskConfig` from
```
task.setCompilerArgs(rcCompiler.getArgs());
```
to
```
task.getCompilerArgs().set(rcCompiler.getArgs());
```

Comment (fletcher-sumglobal) [reply subprojects/language-native/src/main/java/org/gradle/language/rc/tasks/WindowsResourceCompile.java]: Ok, I removed the setter per your instructions.  That is interesting chaining the setter to the getter.

**Comment by big-guy** [comment]
Thanks, this LGTM. I'll merge when CI is green.

**Comment by fletcher-sumglobal** [comment]
@big-guy I noticed some checks were not successful.  Did you pull in all commits for this PR?  There was an unused import statement that I committed after removal of the setter.  I tried to make it all one commit, but would have had to re-branch, and re-commit as one check-in.  Sorry about not having it all done right in the first commit.

**Comment by big-guy** [comment]
Your changes are fine.  The ❌ is from the branch this is merging into.  There were some failures earlier today in master.

**Pull request merged**